lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivityFilters-test.js

```diff
@@ -0,0 +1,272 @@
+      <Config.Provider value={fakeConfig}>
+        <OrganizationActivityFilters
+          selectedCourses={selectedCourses}
+          onCoursesChange={onCoursesChange}
+          selectedAssignments={selectedAssignments}
+          onAssignmentsChange={onAssignmentsChange}
+          selectedStudents={selectedStudents}
+          onStudentsChange={onStudentsChange}
+        />
+      </Config.Provider>,
+    );
+    wrappers.push(wrapper);
+
+    return wrapper;
+  }
+
+  function getSelect(wrapper, id) {
+    return wrapper.find(`MultiSelect[data-testid="${id}"]`);
+  }
+
+  function getSelectContent(wrapper, id) {
+    const content = getSelect(wrapper, id).prop('buttonContent');
+    // Wrap content in div, as it could be a fragment with multiple children
+    // or a scalar value, which would make the call to `mount` fail
+    return mount(<div>{content}</div>).text();
+  }
+
+  it('shows loading indicators while loading', () => {
+    configureFakeAPIFetch(fakeUseAPIFetch, true);
+
+    const wrapper = createComponent();
+
+    assert.equal(getSelectContent(wrapper, 'courses-select'), '...');
+    assert.equal(getSelectContent(wrapper, 'assignments-select'), '...');
+    assert.equal(getSelectContent(wrapper, 'students-select'), '...');
+  });
+
+  it('shows placeholders when selection is empty', () => {
+    const wrapper = createComponent();
+
+    assert.equal(getSelectContent(wrapper, 'courses-select'), 'All courses');
+    assert.equal(
+      getSelectContent(wrapper, 'assignments-select'),
+      'All assignments',
+    );
+    assert.equal(getSelectContent(wrapper, 'students-select'), 'All students');
+  });
+
+  [
+    {
+      id: 'courses-select',
+      expectedOptions: ['All courses', ...courses.map(c => c.title)],
+    },
+    {
+      id: 'assignments-select',
+      expectedOptions: ['All assignments', ...assignments.map(a => a.title)],
+    },
+    {
+      id: 'students-select',
+      expectedOptions: ['All students', ...students.map(s => s.display_name)],
+    },
+  ].forEach(({ id, expectedOptions }) => {
+    it('renders corresponding options', () => {
+      const wrapper = createComponent();
+      const select = getSelect(wrapper, id);
+      const options = select.find(MultiSelect.Option);
+
+      assert.equal(options.length, expectedOptions.length);
+      options.forEach((option, index) => {
+        assert.equal(option.text(), expectedOptions[index]);
+      });
+    });
+  });
+
+  [
+    {
+      id: 'courses-select',
+      getExpectedCallback: () => onCoursesChange,
+    },
+    {
+      id: 'assignments-select',
+      getExpectedCallback: () => onAssignmentsChange,
+    },
+    {
+      id: 'students-select',
+      getExpectedCallback: () => onStudentsChange,
+    },
+  ].forEach(({ id, getExpectedCallback }) => {
+    it('invokes corresponding change callback', () => {
+      const wrapper = createComponent();
+      const select = getSelect(wrapper, id);
+
+      select.props().onChange();
```

Comment:
This checks that the right callback is invoked, but doesn't check the arguments.

Reply:
I thought about that, but since we are calling `onChange` inline, the right callback will be called with whatever we pass there, therefore, not adding a lot of value to check the argument.

I could try to make this more of a functional test, not mocking other components via `$imports.$mock(mockImportedComponents());`, and then try to actually select and item from the listbox, then check that item is passed to the callback, but I think that's not the approach we usually follow.

What do you think?